entity-service-change-event-generator/src/test/java/org/hypertrace/entity/service/change/event/impl/EntityChangeEventGeneratorImplTest.java

```diff
@@ -0,0 +1,156 @@
+package org.hypertrace.entity.service.change.event.impl;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.junit.jupiter.api.Assertions.fail;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+
+import java.util.List;
+import java.util.UUID;
+import java.util.concurrent.atomic.AtomicInteger;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
+import org.hypertrace.core.eventstore.EventProducer;
+import org.hypertrace.core.eventstore.EventStore;
+import org.hypertrace.entity.change.event.v1.EntityChangeEventKey;
+import org.hypertrace.entity.change.event.v1.EntityChangeEventValue;
+import org.hypertrace.entity.data.service.v1.Entity;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.ArgumentCaptor;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+
+@ExtendWith(MockitoExtension.class)
+class EntityChangeEventGeneratorImplTest {
+
+  private static final String TEST_ENTITY_TYPE = "test-entity-type";
+  private static final String TEST_TENANT_ID = "test-tenant-1";
+
+  @Mock EventStore eventStore;
+
+  @Mock EventProducer<EntityChangeEventKey, EntityChangeEventValue> eventProducer;
+
+  EntityChangeEventGeneratorImpl changeEventGenerator;
+
+  @BeforeEach
+  void setup() {
+    changeEventGenerator = new EntityChangeEventGeneratorImpl(eventStore, eventProducer);
+  }
+
+  @Test
+  void sendCreateNotification() {
+    List<Entity> entities = createEntities(2);
+    ArgumentCaptor<EntityChangeEventKey> keyCaptor =
```

Comment:
why use captors rather than plain matchers?

Reply:
Maybe I'm forgetting my mockito, I thought you could just as easily do something like below, which is more readable IMO
```java
verify(eventProducer, times(1)).send(eq(expectedFirstKey), eq(expectedFirstValue));
verify(eventProducer, times(1)).send(eq(expectedSecondKey), eq(expectedSecondValue));
```

Reply:
This will work fine as well. ArgumentCaptor allows to validate order as well.

Reply:
verify supports that too - https://github.com/hypertrace/java-grpc-utils/blob/cb552a00c4967fee19000386f662bfaec7e11f1c/grpc-client-utils/src/test/java/org/hypertrace/core/grpcutils/client/GrpcChannelRegistryTest.java#L96-L103 for example.


Anyway, this was just a nit for readability.

Reply:
Anyways removed capturs. In some cases, it was adding just additional complexity.